docs/concepts/stateful-stream-processing.md

```diff
@@ -123,7 +123,7 @@ provided by Flink's connectors.
 snapshots, we use the words *snapshot* and *checkpoint* interchangeably. Often
 we also use the term *snapshot* to mean either *checkpoint* or *savepoint*.
 
-### Checkpointing
+### Aligned Checkpointing
 
 The central part of Flink's fault tolerance mechanism is drawing consistent
```

Comment:
To my understanding, there are were papers: one by Lamport and one by [DataArtisans](https://arxiv.org/pdf/1506.08603.pdf).
One of the differences is that the former proposed to persist channel state, while the latter proposed alignment (I might be wrong).
So it's probably better to move reference to the paper by Lamport to Unaligned section and mention "Flink" paper in this Aligned section.

Reply:
Good question. I briefly looked at it and it seems as if you are right. I pinged @StephanEwen for clarification and he responded that it is indeed closer, but there are still some difference:

> Flink checkpoints start at sources, propagate through a DAG and persist in-flight as needed
Chandy-Lamport assumes no DAG, starts everywhere at the same time (think: RPC goes to every operators) and all operators log until they saw all markers.

I'd probably leave the reference as is and add a clarifying line to unaligned section.